system/docker/node/Dockerfile

```diff
@@ -1,76 +1,81 @@
+RUN : ${PYTHON3_PYZMQ_VERSION:?"PYTHON3_PYZMQ_VERSION must be provided"}
+ENV PYTHON3_PYZMQ_VERSION=${PYTHON3_PYZMQ_VERSION}
+
 ARG INDY_PLENUM_VERSION
+RUN : ${INDY_PLENUM_VERSION:?"INDY_PLENUM_VERSION must be provided"}
+ENV INDY_PLENUM_VERSION=${INDY_PLENUM_VERSION}
+
 ARG INDY_NODE_VERSION
-ENV INDY_NODE_REPO_COMPONENT=${INDY_NODE_REPO_COMPONENT:-rc}
-ENV LIBINDY_CRYPTO_VERSION=${LIBINDY_CRYPTO_VERSION:-0.4.5}
-ENV PYTHON3_LIBINDY_CRYPTO_VERSION=${PYTHON3_LIBINDY_CRYPTO_VERSION:-0.4.5}
-ENV PYTHON3_PYZMQ_VERSION=${PYTHON3_PYZMQ_VERSION:-18.1.0}
-ENV INDY_PLENUM_VERSION=${INDY_PLENUM_VERSION:-1.12.2}
-ENV INDY_NODE_VERSION=${INDY_NODE_VERSION:-1.12.2~rc1}
-# ENV INDY_NODE_REPO_COMPONENT=${INDY_NODE_REPO_COMPONENT:-master}
-# ENV LIBINDY_CRYPTO_VERSION=${LIBINDY_CRYPTO_VERSION:-0.4.5}
-# ENV PYTHON3_LIBINDY_CRYPTO_VERSION=${PYTHON3_LIBINDY_CRYPTO_VERSION:-0.4.5}
-# ENV PYTHON3_PYZMQ_VERSION=${PYTHON3_PYZMQ_VERSION:-18.1.0}
-# ENV INDY_PLENUM_VERSION=${INDY_PLENUM_VERSION:-1.13.0~dev1012}
-# ENV INDY_NODE_VERSION=${INDY_NODE_VERSION:-1.13.0~dev1199}
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CE7709D068DB5E88 \
-    && echo "deb https://repo.sovrin.org/deb xenial ${INDY_NODE_REPO_COMPONENT}" >> /etc/apt/sources.list \
-    && apt-get update -y && apt-get install -y \
-        libindy-crypto=${LIBINDY_CRYPTO_VERSION} \
-        python3-indy-crypto=${PYTHON3_LIBINDY_CRYPTO_VERSION} \
-        python3-pyzmq=${PYTHON3_PYZMQ_VERSION} \
-        indy-plenum=${INDY_PLENUM_VERSION} \
-        indy-node=${INDY_NODE_VERSION} \
-	&& rm -rf /var/lib/apt/lists/*
+RUN : ${INDY_NODE_VERSION:?"INDY_NODE_VERSION must be provided"}
+ENV INDY_NODE_VERSION=${INDY_NODE_VERSION}
 
+RUN apt-get install -y \
+    libindy-crypto=${LIBINDY_CRYPTO_VERSION} \
+    python3-indy-crypto=${PYTHON3_LIBINDY_CRYPTO_VERSION} \
+    python3-pyzmq=${PYTHON3_PYZMQ_VERSION} \
+    indy-plenum=${INDY_PLENUM_VERSION} \
+    indy-node=${INDY_NODE_VERSION}
 # sovrin & token plugins
-# FIXME
-#   - for now it's tricky to set proper versions for master artifacts
-#   - defaults are for stable
-ARG TOKEN_PLUGINS_INSTALL
+ARG SOVRIN_INSTALL
 ARG SOVRIN_VERSION
-ARG SOVTOKEN_VERSION
-ARG SOVTOKENFEES_VERSION
-ENV TOKEN_PLUGINS_INSTALL=$TOKEN_PLUGINS_INSTALL
+ENV SOVRIN_INSTALL=$SOVRIN_INSTALL
 ENV SOVRIN_VERSION=${SOVRIN_VERSION:-1.1.69}
-ENV SOVTOKEN_VERSION=${SOVTOKEN_VERSION:-1.0.8~dev151}
-# ENV SOVRIN_VERSION=${SOVRIN_VERSION:-1.1.207}
-# ENV SOVTOKEN_VERSION=${SOVTOKEN_VERSION:-1.0.8~dev150}
-ENV SOVTOKENFEES_VERSION=${SOVTOKENFEES_VERSION:-${SOVTOKEN_VERSION}}
-#RUN if [ "$TOKEN_PLUGINS_INSTALL" = "yes" ]; then \
-RUN \
-        set -ex; \
-        apt-get update && apt-get install -y \
-            sovrin=${SOVRIN_VERSION} \
-            sovtoken=${SOVTOKEN_VERSION} \
-            sovtokenfees=${SOVTOKENFEES_VERSION}; \
-        rm -rf /var/lib/apt/lists/*; \
+COPY . . 
+RUN if [ -n "$SOVRIN_INSTALL" ]; then \
+    dpkg -i "sovrin_${SOVRIN_VERSION}_amd64.deb"; \
```

Comment:
this is what installs the pending sovrin package in the GHA.

alternative would be to publish a dev sovrin deb every time a pull request is created, but not sure if that will be something the sovrin-foundation nexus would/should support @WadeBarnes

Reply:
The comment below says that we gonna cache both images, but can we cache the node image since it includes the built source?

Reply:
the github actions responsible for caching can cache based on a hash of changed files. The hash can include multiple file patterns. So if we want node to rebuild whenever the source changes, we could do something like
**/Dockerfile, **.*.py

Reply:
Ok, fine. I just think .py files will be changed in most commits so the cache can make the build process even slower in this case.

Reply:
But anyway, it's fine for the first version.